mono/utils/strenc-internals.h

```diff
@@ -0,0 +1,8 @@
+#ifndef _MONO_STRENC_INTERNALS_H_
+#define _MONO_STRENC_INTERNALS_H_
+
+#include <glib.h>
+
+gchar *mono_unicode_to_external_error (const gunichar2 *uni, GError **err);
```

Comment:
I don't think we should expose `GError*` to the callers.

We should make `mono_unicode_to_external_checked (const guinchar2 *uni, MonoError *error)` and convert the `GError*` into a `MonoError` in its implementation.